packages/datadog-plugin-jest/test/circus.spec.js

```diff
@@ -61,28 +58,18 @@ describe('Plugin', function () {
         .get('/')
         .reply(200, 'OK')
 
-      const loadArguments = [['jest', 'http']]
-
-      const isAgentlessTest = this.currentTest.parent.title === 'agentless'
-      const isEvpProxyTest = this.currentTest.parent.title === 'evp proxy'
-
-      // we need the ci visibility init for the coverage test
-      if (isAgentlessTest) {
-        process.env.DD_API_KEY = 'key'
-        process.env.DD_APP_KEY = 'app-key'
-        process.env.DD_ENV = 'ci'
-        process.env.DD_CIVISIBILITY_ITR_ENABLED = 1
-        process.env.DD_SITE = 'datad0g.com'
-        loadArguments.push({ service: 'test', isIntelligentTestRunnerEnabled: true })
-        loadArguments.push({ experimental: { exporter: 'datadog' } })
-      } else if (isEvpProxyTest) {
-        loadArguments.push({ service: 'test' })
-        loadArguments.push({ experimental: { exporter: 'agent_proxy' } })
-      } else {
-        loadArguments.push({ service: 'test' })
+      process.env.DD_API_KEY = 'key'
+      process.env.DD_APP_KEY = 'app-key'
+      const isAgentlessTest = this.currentTest.parent.title === 'reporting through agentless'
```

Comment:
Relying on a specific test structure like this will make refactoring a bit of a pain in the future. It would be better if this could be handled differently somehow, like function parameters rather than trying to walk up the test tree and compare names.

Reply:
agreed. I just don't know of any other way to do this 😓 . The `agent.load` needs to happen in the global `beforeEach`, and that happens before any `beforeEach` you can put in a `describe` afterwards. So by the time you're in the test you can't change the exporter

Reply:
I found a way to improve it a bit. It isn't great either but should make it clearer

Reply:
fixed in https://github.com/DataDog/dd-trace-js/pull/2584/commits/b7ffb778d8e282067ea59ba73348099523d68d9c